test/unit/unittest_utils/galaxy_mock.py

```diff
@@ -63,22 +66,47 @@ def __init__( self, config=None, **kwargs ):
         self.tag_handler = tags.GalaxyTagManager( self.model.context )
         self.quota_agent = quota.QuotaAgent( self.model )
         self.init_datatypes()
+        self.job_config = Bunch(
+            dynamic_params=None,
+        )
+        self.tool_data_tables = {}
+        self.dataset_collections_service = None
+        self.container_finder = NullContainerFinder()
+        self._toolbox_lock = MockLock()
+        self.genome_builds = GenomeBuilds( self )
+        self.job_queue = NoopQueue()
 
     def init_datatypes( self ):
         datatypes_registry = registry.Registry()
         datatypes_registry.load_datatypes()
         model.set_datatypes_registry( datatypes_registry )
+        self.datatypes_registry = datatypes_registry
+
+    def wait_for_toolbox_reload(self, toolbox):
+        # TODO: If the tpm test case passes, does the operation really
+        # need to wait.
+        return True
+
+
+class MockLock( object ):
+    def __enter__(self):
+        pass
+
+    def __exit__(self, type, value, traceback):
+        pass
 
 
 class MockAppConfig( Bunch ):
 
     def __init__( self, root=None, **kwargs ):
         Bunch.__init__( self, **kwargs )
+        root = root or '/tmp'
```

Comment:
We don't ... or at least shouldn't I don't think ... write anything here - I think it just needs to be a path that exists for the test cases. `pwd` doesn't quite work because it is an actual Galaxy root so it might mess with the tests a bit? I'm not sure to be honest but I'd like to avoid creating a directory for each test if we can to keep them as fast as possible.